src/main/java/io/cdap/plugin/http/common/BaseHttpConfig.java

```diff
@@ -369,16 +400,13 @@ public Header getAuthorizationHeader() throws IOException {
 
         switch (authType) {
             case OAUTH2:
-                String accessToken = OAuthUtil.getAccessTokenByRefreshToken(HttpClients.createDefault(), getTokenUrl(),
-                        getClientId(), getClientSecret(),
-                        getRefreshToken());
-                return new BasicHeader("Authorization", "Bearer " + accessToken);
+                this.credentials = OAuthUtil.getUserCredentials(this);
```

Comment:
it's not good to make the config stateful like this, it makes it hard to reason about the state and easy to introduce bugs. It's better to have this return the credential instead of setting a value.